### ISSUE TITLE
Binding Parameters/Inserting into table is not working 

### ISSUE THREAD

**Issue by CPavanKishore** [snowflakedb/snowflake-connector-net#61]
### Issue description
When i tried inserting into table it started throwing object reference error Please Clear this issue or guide me how to 

### Example code
```cs
```

### Error log
```
[2018-09-04 23:06:49,379] [1] [INFO ] [SnowFlake_Core_Test.Program] Application - Main is invoked
[2018-09-04 23:07:10,074] [8] [DEBUG] [Snowflake.Data.Client.SnowflakeDbCommand] Constucting SnowflakeDbCommand class
[2018-09-04 23:07:10,082] [8] [DEBUG] [Snowflake.Data.Client.SnowflakeDbConnection] Open Connection.
[2018-09-04 23:07:10,133] [8] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Start parsing connection string.
[2018-09-04 23:07:10,150] [8] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: SCHEME, value: https
[2018-09-04 23:07:10,150] [8] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: HOST, value: juniper.snowflakecomputing.com
[2018-09-04 23:07:10,154] [8] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: PORT, value: 443
[2018-09-04 23:07:10,155] [8] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: USER, value: XXXXXXX
[2018-09-04 23:07:10,155] [8] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: PASSWORD, value: XXXXXXXX
[2018-09-04 23:07:10,155] [8] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: ACCOUNT, value: juniper
[2018-09-04 23:07:10,155] [8] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: ROLE, value: 
[2018-09-04 23:07:10,155] [8] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: DB, value: XXXXXXXX
[2018-09-04 23:07:10,155] [8] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: SCHEMA, value: DEV
[2018-09-04 23:07:10,155] [8] [INFO ] [Snowflake.Data.Core.SFSessionProperties] Connection property: WAREHOUSE, value: XXXXXX
[2018-09-04 23:07:10,161] [8] [DEBUG] [Snowflake.Data.Core.SFSessionProperties] Sesssion property CONNECTION_TIMEOUT set to default value: 0
[2018-09-04 23:07:10,163] [8] [DEBUG] [Snowflake.Data.Core.SFSession] Open Session
[2018-09-04 23:07:12,582] [8] [DEBUG] [Snowflake.Data.Core.SFSession] Login Request Data: SFRestRequest {url: https://juniper.snowflakecomputing.com/session/v1/login-request?warehouse=LEGAL_WH&databaseName=EDP_COMPLIANCE&schemaName=DEV&roleName=&requestId=9ea9f7ae-149f-487e-9d05-f321eac5c653, request body: AuthRequest {data: AuthRequestData {ClientAppVersion: 1.0.3.0 AccountName: juniper, loginName: ETL_COMPLIANCE, ClientEnv: { APPLICATION: dotnet, OS_VERSION: Microsoft Windows NT 6.2.9200.0, NET_RUNTIME: .NET Core 4.6.26328.01 } } } }
[2018-09-04 23:07:12,652] [10] [DEBUG] [Snowflake.Data.Core.RestRequestImpl] Http method: POST, http request message: Method: POST, RequestUri: 'https://juniper.snowflakecomputing.com/session/v1/login-request?warehouse=XXXXX&databaseName=XXXXX&schemaName=XXXX&roleName=&requestId=9ea9f7ae-149f-487e-9d05-f321eac5c653', Version: 1.1, Content: System.Net.Http.StringContent, Headers:
{
  Authorization: Basic
  Accept: application/snowflake
  Content-Type: application/json; charset=utf-8
}
[2018-09-04 23:07:15,126] [9] [DEBUG] [Snowflake.Data.Core.HttpUtil+RetryHandler] Success Response: StatusCode: 200, ReasonPhrase: 'OK', Version: 1.1, Content: System.Net.Http.NoWriteNoSeekStreamContent, Headers:
{
  Connection: keep-alive
  Date: Tue, 04 Sep 2018 17:37:14 GMT
  Server: nginx
  Strict-Transport-Security: max-age=31536000
  X-Frame-Options: deny
  Content-Length: 2860
  Content-Type: application/json
}
[2018-09-04 23:07:15,134] [9] [DEBUG] [Snowflake.Data.Core.RestRequestImpl] Post response: {
  "data" : {
  "masterToken" : "ETMsDgAAAWWlqOfHABRBRVMvQ0JDL1BLQ1M1UGFkZGluZwCAABAAEHq3hnxdNqlcxVxwd18q5Z0AAACgAdHGiIH0XkVNV6RMuQq70zPf/NMNwI1+OZOablohoyUJanYe9JXPJkFyYCJu6g/EaegX7/RwOoTOP0i8FUskNDQVxFf2Eu5pQna0L0RmR+6vYC/E2ixI4lfJbCv+qf3dRFwHVaoKw9fv46mKaHqJei0npUEyX5b3izGaoWPXX68XKVlqufRYsXraVEvzVqGPmoFdFkHH2K8s0Wu/Qbh/4QAUqd9nUlClZV/mHLrVHn+GkQSMJNg=",
  "token" : "ETMsDgAAAWWlqOfXABRBRVMvQ0JDL1BLQ1M1UGFkZGluZwCAABAAEBPuLxY+QfTgp5OkdAr91poAAACATtU1aHXGWdTpSe9UMeFn5KgrMr/y7ozTqZdRbSzwE7M9zVHSzli8Vp8rvqIPzuoCVvsreo0K/jzkrlMSsiCgdl3JYK+fXn5/GQQYSs6WWLQTVJbzdxOJ22n9c2m1aZ6konPk7yblOlAqyhI9Sv0UBEMuyKXm/O7wXehC0Ih6g+gAFN8Kg6uaqdAi5OkrTLccmXNzxWWL",
  "validityInSeconds" : 3600,
  "masterValidityInSeconds" : 14400,
  "displayUserName" : "Compliance ETL User",
  "serverVersion" : "2.56.0",
  "firstLogin" : false,
  "remMeToken" : null,
  "remMeValidityInSeconds" : 0,
  "healthCheckInterval" : 45,
  "newClientForUpgrade" : null,
  "sessionId" : 21427593734134,
  "parameters" : [ {
    "name" : "TIMEZONE",
    "value" : "America/Los_Angeles"
  }, {
    "name" : "TIMESTAMP_OUTPUT_FORMAT",
    "value" : "YYYY-MM-DD HH24:MI:SS.FF3 TZHTZM"
  }, {
    "name" : "TIMESTAMP_NTZ_OUTPUT_FORMAT",
    "value" : "YYYY-MM-DD HH24:MI:SS.FF3"
  }, {
    "name" : "TIMESTAMP_LTZ_OUTPUT_FORMAT",
    "value" : ""
  }, {
    "name" : "TIMESTAMP_TZ_OUTPUT_FORMAT",
    "value" : ""
  }, {
    "name" : "DATE_OUTPUT_FORMAT",
    "value" : "YYYY-MM-DD"
  }, {
    "name" : "TIME_OUTPUT_FORMAT",
    "value" : "HH24:MI:SS"
  }, {
    "name" : "CLIENT_DISABLE_INCIDENTS",
    "value" : false
  }, {
    "name" : "CLIENT_HONOR_CLIENT_TZ_FOR_TIMESTAMP_NTZ",
    "value" : true
  }, {
    "name" : "CLIENT_RESULT_PREFETCH_SLOTS",
    "value" : 2
  }, {
    "name" : "CLIENT_RESULT_PREFETCH_THREADS",
    "value" : 1
  }, {
    "name" : "CLIENT_USE_V1_QUERY_API",
    "value" : true
  }, {
    "name" : "CLIENT_SESSION_KEEP_ALIVE",
    "value" : false
  }, {
    "name" : "CLIENT_TELEMETRY_ENABLED",
    "value" : true
  }, {
    "name" : "BINARY_OUTPUT_FORMAT",
    "value" : "HEX"
  }, {
    "name" : "AUTOCOMMIT",
    "value" : true
  }, {
    "name" : "CLIENT_PREFETCH_THREADS",
    "value" : 4
  }, {
    "name" : "CLIENT_MEMORY_LIMIT",
    "value" : 1536
  }, {
    "name" : "CLIENT_METADATA_REQUEST_USE_CONNECTION_CTX",
    "value" : false
  }, {
    "name" : "CLIENT_TIMESTAMP_TYPE_MAPPING",
    "value" : "TIMESTAMP_LTZ"
  }, {
    "name" : "CLIENT_SESSION_CLONE",
    "value" : false
  }, {
    "name" : "CLIENT_STAGE_ARRAY_BINDING_THRESHOLD",
    "value" : 0
  } ],
  "sessionInfo" : {
    "databaseName" : "EDP_COMPLIANCE",
    "schemaName" : "DEV",
    "warehouseName" : "XXXXX",
    "roleName" : "XXXXX"
  },
  "idToken" : null,
  "idTokenValidityInSeconds" : 0,
  "responseData" : null
},
  "message" : null,
  "code" : null,
  "success" : true
}
[2018-09-04 23:07:15,238] [8] [DEBUG] [Snowflake.Data.Core.SFSession] Update parameter map
[2018-09-04 23:07:15,243] [8] [DEBUG] [Snowflake.Data.Client.SnowflakeDbCommand] ExecuteNonQuery, command: insert into EDP_COMPLIANCE.DEV.TC_PROFILE_FORM values (?),(?)
```

### Configuration
*Driver version:*

*Dotnet framework and version:* .Net Core

*Server version:* E.g. 1.90.1
You may get the server version by running a query:
```
SELECT CURRENT_VERSION();
```

*Client OS:* Windows 10

**Comment by howryu** [comment]
Can you give me a code example on how to reproduce it? Sorry, it seemed that my logging is too primitive and did not have enough information. Does it relates to permission issue?

**Comment by CPavanKishore** [comment]
No it's not a permission issue, I found two actually one is If I send null values in between other parameter values it is throws an error and other is object reference error in SFStatement.cs BuildQueryRequest

**Comment by ChTimTsubasa** [comment]
This is potentially fixed and will be in 1.0.10. Please keep an eye on it.

**Comment by sfc-gh-igarish** [comment]
To clean up and re-prioritize more pressing bugs and feature requests we are closing all issues older than 6 months as of April 1, 2023. If there are any issues or feature requests that you would like us to address, please create them according to the new templates we have created. For urgent issues, opening a support case with this link [Snowflake Community](https://community.snowflake.com/s/article/How-To-Submit-a-Support-Case-in-Snowflake-Lodge) is the fastest way to get a response.
